rskj-core/src/main/java/co/rsk/net/MessageCounter.java

```diff
@@ -0,0 +1,42 @@
+package co.rsk.net;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Keeps the counter of messages per peer
+ */
+public class MessageCounter {
+
+    private Map<NodeID, AtomicInteger> messagesPerNode = new HashMap<>();
+
+    private static final AtomicInteger ZERO = new AtomicInteger(0);
+			
+    
+    public int getValue(Peer sender) {
+    	return Optional.ofNullable(messagesPerNode.get(sender.getPeerNodeID())).orElse(ZERO).intValue();
+    }
+    
+    public void increment(Peer sender) {
+    	messagesPerNode.computeIfAbsent(sender.getPeerNodeID(), this::createAtomicInteger);
```

Comment:
`computeIfAbsent` already returns a value, so probably it's no need to call `get` on `messagesPerNode` one more time

Reply:
true